azext_iot/operations/hub.py

```diff
@@ -2434,6 +2434,96 @@ def iot_hub_distributed_tracing_update(
     )
 
 
+def iot_hub_connection_string_show(
+    cmd,
+    hub_name=None,
+    resource_group_name=None,
+    policy_name='iothubowner',
+    key_type=KeyType.primary.value,
+    show_all=False,
+    include_eventhub=False
+):
+    discovery = IotHubDiscovery(cmd)
+
+    if hub_name is None:
+        hubs = discovery.get_iothubs(resource_group_name)
+        if hubs is None:
+            raise CLIError("No IoT Hub found.")
+
+        def conn_str_getter(hub):
+            return _get_hub_connection_string(
+                discovery,
+                hub,
+                policy_name,
+                key_type,
+                show_all,
+                include_eventhub
+            )
+        return [{'name': hub.name, 'connectionString': conn_str_getter(hub)} for hub in hubs]
```

Comment:
The return format when not specifying a hub is not consistent with the return format when given a target hub. In this case `connectionString` value is always a list.

If there is a target hub, then `connectionString` becomes a string value unless `--show-all` is provided in which case it becomes a list.

Reply:
Fixed